lib/bible_bot/parser.rb

```diff
@@ -1,48 +1,31 @@
 module BibleBot
-  
+
   #From forgeapps/scripture_parser v0.0.1
   class Parser
-  
-    def get_book(name)
-      # Get a book from its name or None if not found
-    
-      Bible.books.each do |book|
-        if name.match(Regexp.new(book.regex, Regexp::IGNORECASE))
-          return book
-        end
-      end
-      return nil
-    end
-  
-    def get_book_by_id(id)
-      # Get a book from its id or None if not found
-      Bible.books.each do |book|
-        if id == book.id
-          return book
-        end
-      end
-      return nil
-    end
-  
     def extract(text)
-      #Extract a list of tupled scripture references from a block of text
+      #Extract an array of scripture references from a block of text
       references = []
       text.scan(Bible.new.scripture_re).each do |match|
+
+        # Skip results where bookname is nil
+        bookname = match[0]
+        next if bookname == "" || bookname.nil?
```

Comment:
I think this might have been an issue because of the trailing `|` in the list of books mentioned in a previous comment, but either way, if there is no bookname match, we should skip because it isn't a valid reference.